IPNI_HTTP_PROVIDER.md

```diff
@@ -0,0 +1,310 @@
+- **`sig`** (required): The signature associated with the `head` CID, obtained by concatenating the bytes of the `head`
+  CID and the UTF-8 bytes of the `topic` (if present). The signature is verified against the `pubkey`.
+
+Please note that the response provides the necessary information to validate the authenticity of the advertisement CID
+and verify its integrity using the provider's public key and the associated signature.
+
+The following snippet represents the IPLD schema of the signed head advertisement:
+
+```ipldsch
+type SignedHead  struct {
+    head   Link
+    topic  optional String
+    pubkey Bytes
+    sig    Bytes
+}
+```
+
+#### Example
+
+The following represents an abbreviated example of head advertisement response in `DAG-JSON` encoding:
+
+```json
+{
+  "head": {
+    "/": "baguqeerazx6qhbdzckrnuwvl3reqnrpysz3ry5qtdxzpo23losoeiywij65a"
+  },
+  "topic": "/indexer/ingest/devnet",
+  "pubkey": {
+    "/": {
+      "bytes": "CAASpgIwggEiMA0G..."
+    }
+  },
+  "sig": {
+    "/": {
+      "bytes": "LrZiicKdqDqkG2UFR..."
+    }
+  }
+}
+```
+
+### GET `/ipni/v1/ad/{CID}`
+
+Returns the content associated with an advertisement CID and its entries.
+
+### Response
+
+The response from the `/ipni/v1/ad/{CID}` endpoint returns the data associated to the advertisement CID and its embedded links, including `Entries`. See [IPNI Specification/Advertisements](IPNI.md#advertisements) for the advertisement and entries schema. 
+
+#### Example
+
+The following represents and abbreviated advertisement encoded in `DAG-JSON`:
+
+```json
+{
+  "Addresses": [
+    "/dns4/content.example/tcp/443/wss"
+  ],
+  "ContextID": {
+    "/": {
+      "bytes": "YmFndXFlZXJh..."
+    }
+  },
+  "Entries": {
+    "/": "baguqeeraog3uqu7au2mctrfslzvua5lzl3wyoquqhbyoig6hnputypej2uvq"
+  },
+  "IsRm": false,
+  "Metadata": {
+    "/": {
+      "bytes": "gBI"
+    }
+  },
+  "PreviousID": {
+    "/": "baguqeeranqfhpdacbe7ls44ndcgeis6yufvhk4zntydlqfaplzz6mdyuu6gq"
+  },
+  "Provider": "QmQzqxhK82kAmK...",
+  "Signature": {
+    "/": {
+      "bytes": "CqsCCAASpgIwggEiMA0..."
+    }
+  }
+}
+```
+
+The following snippet shows an example of advertisement entries encoded in `DAG-JSON`: 
+
+```json
+{
+  "Entries": [
+    {
+      "/": {
+        "bytes": "EiDYa/LrNYDdHjSlKGXpHjc1IyHds9Xdi/p8d25Q5UE/bQ"
+      }
+    }
+  ]
+}
+```
+
+### Security
+
+The IPNI HTTP Provider API specification includes several security implications and considerations to ensure secure
+communication and data integrity. Here are the key security aspects:
+
+- **TLS (Transport Layer Security)**: It is recommended to use TLS whenever possible to establish secure connections
+  between clients and the IPNI HTTP provider. TLS encrypts the data during transmission, protecting it from unauthorized
+  access and tampering.
+
+- **Pagination and Maximum Content Length**: Implementers should consider implementing pagination mechanisms and setting
```

Comment:
Don't you want to standardize how paging is communicated between clients and servers to avoid divergence between implementations?

Reply:
Pagination is achieved by IPLD links across entry chunks. I'll add a paragrah to spell this out. Thanks Steve.